src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -27,4 +27,44 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object object) {
+        if (this == object) {
+            return true;
+        }
+
+        if (object == null) {
+            return false;
+        }
+
+        if (getClass() != object.getClass()) {
+            return false;
+        }
+
+        Rectangle rectangle = (Rectangle) object;
+        if ((getWidth() == null && rectangle.getWidth() == null)
+                || (getWidth() != null && getWidth().equals(rectangle.getWidth()))
+                && (getLength() == null && rectangle.getLength() == null)
+                || (getLength() != null && getLength().equals(rectangle.getLength()))
+                && (getColor() == null && rectangle.getColor() == null)
+                || (getColor() != null && getColor().equals(rectangle.getColor()))) {
+            return true;
+        }
+        return false;
+    }
+
+    @Override
+    public int hashCode() {
+        final int prime = 17;
+        int result = 31;
+
```

Comment:
there are too many blank lines in this method

Reply:
done